Options.py

```diff
@@ -426,6 +428,7 @@ def __ne__(self, other):
 
 class TextChoice(Choice):
     """Allows custom string input and offers choices. Choices will resolve to int and text will resolve to string"""
+    value: typing.Union[str, int]
```

Comment:
This is something that the type checkers don't like.
Inheritance is supposed to be from general to specific ("Animal" to "Cat").
But here, the inheritance is going from something more specific to something more general (`value: int` to `value: str | int`).

Maybe it's not really important for this pull request, because it might take some major rearrangements in the inheritance structure...

Reply:
The only way I think to satisfy that would be to invert `Choice` and `TextChoice`, which I suppose is possible, but it's also a bit strange.

Reply:
It would take at least a little more than that, because `Choice` inherits from `NumericOption`.

Reply:
I think the best way to resolve this would be to remove the existence of `TextChoice`.

 - `Choice` already does everything that I would expect something called `TextChoice` to do.
    - (There is text associated with each choice in `Choice`.)
 - No games use it.
    - (There are a few unused imports in some games.)

The only usage of `TextChoice` is in `PlandoBosses`.
I don't understand what `PlandoBosses` is about, but it seems to me, it either doesn't need to inherit from `Choice` at all, or it could inherit directly from `Choice`.

Reply:
looking at it a little more, I'm back to "some major rearrangements in the inheritance structure"

Reply:
I already use TextChoice in another PR.

Reply:
Yeah, I think I see now what its purpose is. Correct me if I'm wrong.

`Choice` is an enumeration of choices for an option.

For `TextChoice`, we don't want to enumerate all of the choices, but we want to enumerate some of them so they can be displayed in the website settings page select box. In addition to those choices that are enumerated and in the select box on the web page, the player can also put text in the yaml for choices that aren't enumerated.

Is this right?

Reply:
Yes. What I'm specifically using it for is for LTTP's Custom Sprite option. It'll display some basic sprite choices but the user can also specify any valid support sprite name as well. You can also use this for custom in game names or such, with some predetermined names to make it easier on the user but still allow them to enter their own custom name. Also PlandoBosses didn't exist when I wrote this up, and is a very specific (but still generic) implementation of TextChoice.

Reply:
With that understanding, my initial thoughts are:
Whatever is required for the website to put things in a select box on the settings page
should be in a mixin.
Then both `Choice` and `TextChoice` could inherit from that mixin, and then `TextChoice` doesn't inherit from `Choice`.

Again, I don't think this is important for this pull request - just spewing out thoughts on what I see.

Reply:
The reason I'm inheriting from Choice is that I'm still using some of the functionality it has, and especially a ton of the functionality defined by NumericOption. The only way to make it not inherit would require a new Option class that NumericOption and TextChoice would inherit from. Personally, I don't think it's worth the effort, just for a very slight styling thing.

Reply:
> The only way to make it not inherit would require a new Option class that NumericOption and TextChoice would inherit from.

That might be the mixin that I'm talking about, or another mixin.

> just for a very slight styling thing

I think this is not a styling thing. This is an invalid inheritance structure. It could cause problems as more people use these classes for more different things, because not every `TextChoice` is a valid `Choice`.

Reply:
> I don't think it's worth the effort

This, I might agree with. That's why I've said it's not an issue for this pull request. And I'm not saying "We need to fix this right now."